This patch is needed to make
    eval "$(rbenv init -)"
work without complaining about
    -bash: /opt/local/bin/../completions/rbenv.bash: No such file or directory

--- libexec/rbenv-init.orig
+++ libexec/rbenv-init
@@ -95,10 +95,15 @@ fish )
 ;;
 esac
 
-completion="${root}/completions/rbenv.${shell}"
-if [ -r "$completion" ]; then
-  echo "source '$completion'"
-fi
+case "$shell" in
+bash )
+  echo "source \"$root/../etc/bash_completion.d/rbenv\""
+  ;;
+zsh )
+  echo "source \"$root/../share/zsh/site-functions/rbenv.zsh_completion\""
+  ;;
+# support for fish and other shells missing
+esac
 
 if [ -z "$no_rehash" ]; then
   echo 'command rbenv rehash 2>/dev/null'
